### PR TITLE
fix(web): simplify hide transition

### LINKED ISSUE #5907: bug(web): click in OSK area when OSK hidden can retrigger OSK for input elements
**Describe the bug**

On a tablet or phone, when an input element has been focused and then blurred, clicking in the area where the KeymanWeb OSK was visible will reactivate the OSK and clicks the key that then happens to be under the cursor.

~~This does not happen for textarea elements, only for input elements.~~ On my second attempt, I reproduced this happening with textarea as well.

**To Reproduce**

1. Go to unminified test page for KeymanWeb on a touch device (or emulated in Chrome)
2. Click on the input field
3. Click outside the input field to blur and hide the OSK
4. Click anywhere within the rectangle where the OSK had been visible, including on other elements.
5. Up pops the OSK and a character is inserted into the input field. Ooops.

**Expected behavior**
The OSK should stay gone!

**Screenshots**

https://user-images.githubusercontent.com/4498365/141704083-ddfd33dc-79c4-4854-81da-226b7ce0e22e.mp4

---

**KeymanWeb:**
 - OS: touch OSes 
 - KeymanWeb Version: 15.0.146-alpha

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #5907.

The state machine for the Hide transition for the OSK was very complex, and had some peculiarities which caused it to sometimes fail to correctly hide. Once I simplified the state machine, it seems that those edge cases have gone away, and the keyboard always hides correctly.

# User Testing

Note: There is an unrelated issue with focus timeout which can be confusing when testing this; see #5909. **To avoid this:** wait at least one second before clicking outside a text box.

* GROUP_DESKTOP: run this test in Chrome on Windows.
* GROUP_TOUCH: run this test on iOS or Android device.

* TEST_KEYBOARD: in the unminified test page, repeatedly switch between controls and ensure that the OSK shows and hides exactly as expected. Try clicking elsewhere in the document to hide the keyboard. Once the keyboard hides, click where it *was* and make sure that it does not magically re-appear! Basically, the idea is to try and get the keyboard interactions to go wrong.

### CODE REVIEW

File: web/source/osk/oskView.ts [hunk 1]
```diff
@@ -851,68 +850,65 @@ namespace com.keyman.osk {
...
+          _this._Box.removeEventListener('webkitTransitionEnd', cleanup, false);
+          _this._Box.removeEventListener('transitioncancel', cleanup, false);
+          _this._Box.removeEventListener('webkitTransitionCancel', cleanup, false);
+          if(_this._animatedHideTimeout != 0) {
+            window.clearTimeout(_this._animatedHideTimeout);
+          }
+          _this._animatedHideTimeout = 0;
+
+          if(_this._Visible && _this.activationConditionsMet) {
+            // Leave opacity alone and clear transition if another element activated
```
Comment (darcywong00) [inline web/source/osk/oskView.ts:880]:
```suggestion
            // Set opacity and clear transition if another element activated
```

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by MakaraSok** [comment]
* **GROUP_DESKTOP: PASSED** The OSK area doesn't output anything when it is invisible.
* **GROUP_TOUCH: PASSED** The OSK area doesn't output anything when it is invisible.
* **TEST_KEYBOARD: FAILED** The OSK doesn't show/appear robustly.
iPad Simulator: at some points the OSK doesn't reappear when clicking in the empty space.
https://user-images.githubusercontent.com/28331388/141936432-fabdd2f0-f323-4a10-9323-d9bf98738e9d.mov

**Comment by mcdurdin** [comment]
> * At some points the OSK doesn't reappear when clicking in the empty space.

Do you mean 'disappear' not 'reappear'?

This is due to #5909 which I am still working on. **To avoid this problem in testing:** wait at least one second before clicking in the empty space. Hope that makes sense? 

@keymanapp-test-bot retest

File: web/source/osk/oskView.ts (hunk 1 above)
Comment (mcdurdin) [reply web/source/osk/oskView.ts:880]: It's kinda 'reset opacity to original value' ... 

```suggestion
            // Reset opacity to original value and clear transition if another element activated
```

**Comment by MakaraSok** [comment]
* GROUP_DESKTOP: run this test in Chrome on Windows.
  - **TEST_KEYBOARD: PASSED** The keyboard doesn't reappear when clicking in the same blank area.

* GROUP_TOUCH: run this test on iOS or Android device.
  - **TEST_KEYBOARD: PASSED** The keyboard doesn't reappear when clicking in the same blank area.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.154-alpha](https://keyman.com/downloads/releases/alpha/15.0.154)

**Pull request merged**